app/services/shepherd.js

```diff
@@ -0,0 +1,134 @@
+import { schedule, later } from "@ember/runloop";
+import Service, { inject as service } from "@ember/service";
+import { tracked } from "@glimmer/tracking";
+import TOURS from "timed/tours";
+
+export default class Shepherd extends Service {
+  @service tour;
+  @service notify;
+  @service media;
+  @service router;
+  @service autostartTour;
+  @tracked model;
+
+  get routeName() {
+    return this.router.get("currentRoute")?.name;
+  }
+  async prepare() {
+    this.addDefaultStepsOptions();
+    await this.startFromBeginning();
+    await this.showToursOfCurrentRoute();
+    this.watchRouteChanges();
+  }
+
+  async showToursOfCurrentRoute() {
+    if (this.hasTourForRoute()) {
+      const tours = this.getRouteTours();
+      await this.tour.addSteps(tours.map(this.tourToSteps));
+    }
+  }
+
+  async startFromBeginning() {
+    if (this.routeName !== "index.activities.index") {
+      await this.router.transitionTo("index.activities.index");
+    }
+  }
+
+  watchRouteChanges() {
+    this.router.on("routeDidChange", () => {
+      if (this.hasTourForRoute()) {
+        this.tour.hide();
+        this.startTour();
+      }
+    });
+  }
+
+  hasTourForRoute(routeName = false) {
+    return this.autostartTour.tours.includes(routeName || this.routeName);
+  }
```

Comment:
```suggestion
  hasTourForRoute(routeName) {
    return this.autostartTour.tours.includes(routeName || this.routeName);
  }
```
Why the default parameter value?

Reply:
we don't need parameter at all